packages/gestalt/src/Touchable.js

```diff
@@ -73,126 +85,187 @@ const getRoundingStyle = (r: Rounding): Style => {
+    >
+      {children}
+    </div>
+  );
+}
 
-  handleMouseEnter = (event: SyntheticMouseEvent<HTMLDivElement>) => {
-    const { disabled, onMouseEnter } = this.props;
-    if (!disabled && onMouseEnter) {
-      onMouseEnter({ event });
-    }
-  };
+Touchable.propTypes = {
+  accessibilityControls: PropTypes.string,
+  accessibilityExpanded: PropTypes.bool,
+  accessibilityHaspopup: PropTypes.bool,
+  accessibilityLabel: PropTypes.string,
+  children: PropTypes.node,
+  disabled: PropTypes.bool,
+  forwardedRef: PropTypes.oneOfType([
+    PropTypes.func,
+    PropTypes.shape({
+      current: PropTypes.any,
+    }),
+  ]),
+  fullHeight: PropTypes.bool,
+  fullWidth: PropTypes.bool,
+  inline: PropTypes.bool,
+  mouseCursor: PropTypes.oneOf([
+    'copy',
+    'grab',
+    'grabbing',
+    'move',
+    'noDrop',
+    'pointer',
+    'zoomIn',
+    'zoomOut',
+  ]),
+  onBlur: PropTypes.func,
+  onFocus: PropTypes.func,
+  onTouch: PropTypes.func,
+  onMouseEnter: PropTypes.func,
+  onMouseLeave: PropTypes.func,
+  pressBackgroundEnabled: PropTypes.bool,
+  pressScaleEnabled: PropTypes.bool,
+  pressShadowEnabled: PropTypes.bool,
+  rounding: RoundingPropType,
+};
 
-  handleMouseLeave = (event: SyntheticMouseEvent<HTMLDivElement>) => {
-    const { disabled, onMouseLeave } = this.props;
-    if (!disabled && onMouseLeave) {
-      onMouseLeave({ event });
-    }
-  };
+const forwardRef = (props, ref) => <Touchable {...props} forwardedRef={ref} />;
+forwardRef.displayName = 'Touchable';
 
-  handleClick = (event: SyntheticMouseEvent<HTMLDivElement>) => {
-    const { disabled, onTouch } = this.props;
-    if (!disabled && onTouch) {
-      onTouch({ event });
-    }
-  };
-
-  render() {
-    const {
-      accessibilityControls,
-      accessibilityExpanded,
-      accessibilityHaspopup,
-      accessibilityLabel,
-      children,
-      disabled = false,
-      fullWidth = true,
-      fullHeight,
-      mouseCursor = 'pointer',
-      rounding = 0,
-    } = this.props;
-
-    const classes = classnames(
-      styles.touchable,
-      toProps(getRoundingStyle(rounding)).className,
-      {
-        [styles.fullHeight]: fullHeight,
-        [styles.fullWidth]: fullWidth,
-        [styles[mouseCursor]]: !disabled,
-      }
-    );
-
-    return (
-      <div
-        aria-controls={accessibilityControls}
-        aria-disabled={disabled}
-        aria-expanded={accessibilityExpanded}
-        aria-haspopup={accessibilityHaspopup}
-        aria-label={accessibilityLabel}
-        className={classes}
-        onClick={this.handleClick}
-        onBlur={this.handleBlur}
-        onFocus={this.handleFocus}
-        onMouseEnter={this.handleMouseEnter}
-        onMouseLeave={this.handleMouseLeave}
-        onKeyPress={this.handleKeyPress}
-        role="button"
-        tabIndex={disabled ? null : '0'}
-      >
-        {children}
-      </div>
-    );
-  }
-}
+export default React.forwardRef<Props, HTMLDivElement>(forwardRef);
```

Comment:
`React.forwardRef` here is to allow manually setting focus on the Touchable.

This is needed to support focusing on the next item in an item list when user uses the tab key:

![preview-Jun-25-2019_18-53-47](https://user-images.githubusercontent.com/5341184/83976707-7fcbe200-a8b0-11ea-9793-5f1c77168e28.gif)

See https://phabricator.pinadmin.com/D417565 for more details.